pl_examples/test_examples.py

```diff
@@ -28,15 +35,81 @@
     "--data.batch_size 32 "
 )
 ARGS_GPU = ARGS_DEFAULT + "--trainer.gpus 1 "
+ARGS_DP = ARGS_DEFAULT + "--trainer.gpus 2 --trainer.accelerator dp "
+ARGS_AMP = "--trainer.precision 16 "
+
+
+def run(tmpdir, import_cli, cli_args):
+    file = Path(__file__).absolute()
+    cli_args = cli_args % {"tmpdir": tmpdir}
+    # this will execute this exact same file
+    coverage = ["-m", "coverage", "run", "--source", "pytorch_lightning", "-a"]
+    command = [sys.executable, *coverage, str(file), f"--import_cli={import_cli}", f"--cli_args={cli_args}"]
+    p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    try:
+        std, err = p.communicate(timeout=60)
+        print(std)
+        err = str(err.decode("utf-8"))
+        if "Exception" in err or "Error" in err:
+            raise Exception(err)
+    except TimeoutExpired:
+        p.kill()
+        std, err = p.communicate()
+    return std, err
+
+
+# FIXME(@tchaton): Find the reason
+@pytest.mark.skipif(True, reason="this hangs within the CI.")
+@RunIf(min_gpus=2)
+@pytest.mark.parametrize("cli_args", [ARGS_DP, ARGS_DP + ARGS_AMP])
+def test_examples_dp_simple_image_classifier(tmpdir, cli_args):
+    run(tmpdir, "pl_examples.basic_examples.simple_image_classifier", cli_args)
+
+
+# FIXME(@tchaton): Find the reason
+@pytest.mark.skipif(True, reason="this hangs within the CI.")
+@RunIf(min_gpus=2)
+@pytest.mark.parametrize("cli_args", [ARGS_DP, ARGS_DP + ARGS_AMP])
+def test_examples_dp_backbone_image_classifier(tmpdir, cli_args):
+    run(tmpdir, "pl_examples.basic_examples.backbone_image_classifier", cli_args)
+
+
+# FIXME(@tchaton): Find the reason
+@pytest.mark.skipif(True, reason="this hangs within the CI.")
+@RunIf(min_gpus=2)
+@pytest.mark.parametrize("cli_args", [ARGS_DP, ARGS_DP + ARGS_AMP])
+def test_examples_dp_autoencoder(tmpdir, cli_args):
+    run(tmpdir, "pl_examples.basic_examples.autoencoder", cli_args)
+
+
+@pytest.mark.parametrize("cli_args", [ARGS_DEFAULT])
+def test_examples_cpu_simple_image_classifier(tmpdir, cli_args):
+    run(tmpdir, "pl_examples.basic_examples.simple_image_classifier", cli_args)
+
+
+@pytest.mark.parametrize("cli_args", [ARGS_DEFAULT])
+def test_examples_cpu_backbone_image_classifier(tmpdir, cli_args):
+    run(tmpdir, "pl_examples.basic_examples.backbone_image_classifier", cli_args)
+
+
+@pytest.mark.parametrize("cli_args", [ARGS_DEFAULT])
+def test_examples_cpu_autoencoder(tmpdir, cli_args):
+    run(tmpdir, "pl_examples.basic_examples.autoencoder", cli_args)
```

Comment:
This PR not necessary anymore - these are already run here

https://github.com/PyTorchLightning/pytorch-lightning/blob/91ce0d0a99ec0a488e04e09997f6dd662ce217cd/.azure-pipelines/gpu-tests.yml#L108-L112

Perhaps with the exception of the DALI example, but it doesn't seem to be failing, so why change it